frontend/src/common/wot.ts

```diff
@@ -0,0 +1,101 @@
+import { base64 } from 'rfc4648';
+import backend, { TrustDto, UserDto } from './backend';
+import { UserKeys, asPublicKey, getJwkThumbprint } from './crypto';
+import { JWT, JWTHeader } from './jwt';
+import userdata from './userdata';
+
+export type SignedKeys = {
+  ecdhPublicKey: string;
+  ecdsaPublicKey: string;
+}
+
+function deeplyEqual(a: SignedKeys, b: SignedKeys) {
+  return a.ecdhPublicKey === b.ecdhPublicKey
+    && a.ecdsaPublicKey === b.ecdsaPublicKey;
+}
+
+/**
+ * Signs the public key of a user with my private key and sends the signature to the backend.
+ * @param user The user whose keys to sign
+ * @returns The new trust object created during the signing process
+ */
+async function sign(user: UserDto): Promise<TrustDto> {
+  if (!user.ecdhPublicKey || !user.ecdsaPublicKey) {
+    throw new Error('No public key to sign');
+  }
+  const toSign: SignedKeys = {
+    ecdhPublicKey: user.ecdhPublicKey,
+    ecdsaPublicKey: user.ecdsaPublicKey
+  };
+  const me = await userdata.me;
+  const userKeys = await userdata.decryptUserKeysWithBrowser();
+  const signature = await JWT.build({
+    alg: 'ES384',
+    typ: 'JWT',
+    b64: true,
+    iss: me.id,
+    sub: user.id,
+    iat: Math.floor(Date.now() / 1000)
+  }, toSign, userKeys.ecdsaKeyPair.privateKey);
+  await backend.trust.trustUser(user.id, signature);
+  const trust = await backend.trust.get(user.id);
+  return trust!;
```

Comment:
Why this unwrap here? It can be actual undefined due to

https://github.com/cryptomator/hub/blob/99a527c2a1522e3b9cba71eb4886c197d732913c/frontend/src/common/backend.ts#L266C9-L266C57

Reply:
technically `GET` can be 404 but since we just established a trust chain of length 1 by trusting the user directly, we can expect it to be 200.

Therefore I'd consider an undefined trust as erroneous, so better fail hard.